src/js/utils/pagination.js

```diff
@@ -18,24 +18,13 @@ export const usePagination = ({ data, page, step, ...rest }) => {
 
   const itemsBeginIndex = step * (activePage - 1);
   const itemsEndIndex = itemsBeginIndex + step;
-  const getCurrentItems = useCallback(
-    items => {
-      if (Array.isArray(items)) {
-        return items.length
-          ? items.slice(itemsBeginIndex, itemsEndIndex)
-          : undefined;
-      }
-      return items;
-    },
-    [itemsBeginIndex, itemsEndIndex],
-  );
-  const [currentItems, setCurrentItems] = useState(getCurrentItems(data));
 
-  useEffect(() => {
-    if (data) {
-      setCurrentItems(getCurrentItems(data));
+  const currentItems = useMemo(() => {
+    if (Array.isArray(data)) {
+      return data.length ? data.slice(itemsBeginIndex, itemsEndIndex) : [];
```

Comment:
if data is an array of length 0, wont' `data.slice(...)` return `[]` already?

Reply:
You're right. I think we can remove the check for length now